File: backend/internal/model/assignment.go
package model

import "time"

type AssignmentRule struct {
	ID         string         `json:"id"`
	ProjectID  string         `json:"project_id"`
	RuleType   string         `json:"rule_type"`
	RuleConfig map[string]any `json:"rule_config"`
	Priority   int            `json:"priority"`
	IsActive   bool           `json:"is_active"`
	CreatedBy  string         `json:"created_by"`
	CreatedAt  time.Time      `json:"created_at"`
}

type AssignmentCandidate struct {
	UserID        string    `json:"user_id"`
	FullName      string    `json:"full_name"`
	Email         string    `json:"email"`
	AvatarURL     *string   `json:"avatar_url,omitempty"`
	Department    *string   `json:"department,omitempty"`
	Skills        []string  `json:"skills,omitempty"`
	RoleInProject string    `json:"role_in_project"`
	AssignedAt    time.Time `json:"assigned_at"`
	Workload      int       `json:"workload"`
}

File: frontend/src/types/assignment.ts
import type { KanbanTask } from "@/types/kanban";

export type AssignmentRuleType = "by_department" | "by_skill" | "by_workload";

export interface AssignmentRule {
  id: string;
  project_id: string;
  rule_type: AssignmentRuleType;
  rule_config: Record<string, unknown>;
  priority: number;
  is_active: boolean;
  created_by: string;
  created_at: string;
}

export interface AssignmentRuleFormValues {
  rule_type: AssignmentRuleType;
  config_value: string;
  role_in_project: string;
  priority: number;
  is_active: boolean;
}

export interface AutoAssignResult {
  task: KanbanTask;
  matched_rule: AssignmentRule;
  assigned_to: {
    user_id: string;
    full_name: string;
    email: string;
    role_in_project: string;
    workload: number;
  };
}
